Sources/Sentry/SentryEnvelope.m

```diff
@@ -231,15 +232,17 @@ - (nullable instancetype)initWithReplayEvent:(SentryReplayEvent *)replayEvent
                                        video:(NSURL *)videoURL
 {
     NSData *replayEventData = [SentrySerialization dataWithJSONObject:[replayEvent serialize]];
-    NSData *recording = [SentrySerialization dataWithReplayRecording:replayRecording];
+    NSData *_Nullable recording = [SentrySerialization dataWithReplayRecording:replayRecording];
     NSURL *envelopeContentUrl =
         [[videoURL URLByDeletingPathExtension] URLByAppendingPathExtension:@"dat"];
 
-    BOOL success = [SentryMsgPackSerializer serializeDictionaryToMessagePack:@{
-        @"replay_event" : replayEventData,
-        @"replay_recording" : recording,
-        @"replay_video" : videoURL
+    NSMutableDictionary *envelopeContent = [NSMutableDictionary dictionary];
+    envelopeContent[@"replay_event"] = replayEventData;
+    if (nil != recording) {
```

Comment:
Instead of skipping the recording data, it might make sense to abort the envelope initialization instead. WDYT?

Reply:
If a replay event without a recording is still somewhat useful, then I would say keep the envelope with only the replay event. If Relay or the backend discards such envelopes, then we could throw away the envelope here already.